c7n/resources/kms.py

```diff
@@ -62,11 +62,15 @@ def augment(self, resources):
         for r in resources:
             try:
                 key_id = r.get('KeyId')
-                key_arn = r.get('KeyArn', key_id)
-                info = client.describe_key(KeyId=key_arn)['KeyMetadata']
+                # We get `KeyArn` from list_keys and `Arn` from describe_key.
+                # If we already have describe_key info we don't need to fetch
+                # it again.
+                if 'Arn' not in r:
+                    key_arn = r.get('KeyArn', key_id)
+                    info = client.describe_key(KeyId=key_arn)['KeyMetadata']
+                    r.update(info)
                 if key_id in alias_map:
-                    info['AliasNames'] = alias_map[key_id]
-                r.update(info)
+                    r['AliasNames'] = alias_map[key_id]
             except ClientError as e:
                 if e.response['Error']['Code'] == 'AccessDeniedException':
```

Comment:
In the case of `AccessDenied` for a key, this ends up with a resource with just the list-keys output, which includes only the `KeyId` and `KeyArn` fields.  It causes a `KeyError()` exception later because the `kms-key` arn expects `Arn`, which is only available after the describe call is merged in.  See #6696 for example output in this case.  Just wondering if this could/should be addressed as part of these changes or if it should be a separate issue.

Reply:
Oh, I see now that's #6245